src/App.js

```diff
@@ -26,40 +26,92 @@ const generateSquares = () => {
+  };
+
   const [squares, setSquares] = useState(generateSquares());
 
-  // Wave 2
-  // You will need to create a method to change the square
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  const changeSquareValue = (id) => {
+    let newSquares = [];
+    for (let row of squares) {
+      let rowArray = [];
+      for (let column of row) {
+        if (column.id === id && !column.value) {
+          rowArray.push({id: column.id,
+          value: playerTurn});
+          toggleTurn();
+        } else 
+        {rowArray.push(column);}
+      } newSquares.push(rowArray); 
+    }
+    setSquares(newSquares);
+  };
 
   const checkForWinner = () => {
-    // Complete in Wave 3
-    // You will need to:
-    // 1. Go accross each row to see if
-    //    3 squares in the same row match
-    //    i.e. same value
-    // 2. Go down each column to see if
-    //    3 squares in each column match
-    // 3. Go across each diagonal to see if
-    //    all three squares have the same value.
+    let squaresArray = [];
+
+    for (let rowIndex in squares) {
+      for (let objectIndex in squares[rowIndex]) {
+        squaresArray.push(squares[rowIndex][objectIndex].value);
+      }
+    }
+
+    // ROWS
+    if (squaresArray[0] != '' && squaresArray[0] === squaresArray[1] && squaresArray[0] === squaresArray[2]) {
+      return squaresArray[0];
+    } else if (squaresArray[3] != '' && squaresArray[3] === squaresArray[4] && squaresArray[3] === squaresArray[5]) {
+      return squaresArray[3];
+    } else if (squaresArray[6] != '' && squaresArray[6] === squaresArray[7] && squaresArray[6] === squaresArray[8]) {
+      return squaresArray[6];
+    } 
+    // COLUMNS 
+    else if (squaresArray[0] != '' && squaresArray[0] === squaresArray[3] && squaresArray[0] === squaresArray[6]) {
+      return squaresArray[0];
+    } else if (squaresArray[1] != '' && squaresArray[1] === squaresArray[4] && squaresArray[1] === squaresArray[7]) {
+      return squaresArray[1];
+    } else if (squaresArray[2] != '' && squaresArray[2] === squaresArray[5] && squaresArray[2] === squaresArray[8]) {
+      return squaresArray[2];
+    }
+    // DIAGONALS
+    else if (squaresArray[0] != '' && squaresArray[0] === squaresArray[4] && squaresArray[0] === squaresArray[8]) {
+      return squaresArray[0];
+    } else if (squaresArray[6] != '' && squaresArray[6] === squaresArray[4] && squaresArray[6] === squaresArray[2]) {
+      return squaresArray[6];
+    }
+
+    if (squaresArray.includes('')) {
+      return null;
+    } else {
+      return 'Tie';
+    }
+  };
+
+  const declareWinner = () => {
+    if (!checkForWinner()) {
+      return <h2>It&apos;s Player {playerTurn}&apos;s turn!</h2>;
+    } else if (checkForWinner() == 'Tie') {
+      return <h2>Y&apos;all both lose! &#128532;</h2>;
+    } else {
+      return <h2>&#127881; Player {checkForWinner()} wins the game! &#127881;</h2>;
```

Comment:
The test is expecting a lowercase for the winner declaration. Consider how we can change the output to lowercase just here.